smack-experimental/src/main/java/org/jivesoftware/smackx/muclight/MultiUserChatLight.java

```diff
@@ -0,0 +1,555 @@
+     */
+    public MUCLightRoomConfiguration getConfiguration(String version)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        MUCLightGetConfigsIQ mucLightGetConfigsIQ = new MUCLightGetConfigsIQ(room, version);
+        StanzaFilter responseFilter = new IQReplyFilter(mucLightGetConfigsIQ, connection);
+
+        IQ responseIq = connection.createPacketCollectorAndSend(responseFilter, mucLightGetConfigsIQ)
+                .nextResultOrThrow();
+        MUCLightConfigurationIQ mucLightConfigurationIQ = (MUCLightConfigurationIQ) responseIq;
+
+        return mucLightConfigurationIQ.getConfiguration();
+    }
+
+    /**
+     * Get the MUC Light configuration.
+     * 
+     * @return the room configuration
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public MUCLightRoomConfiguration getConfiguration()
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        return getConfiguration(null);
+    }
+
+    /**
+     * Get the MUC Light affiliations.
+     * 
+     * @param version
+     * @return the room affiliations
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public HashMap<Jid, MUCLightAffiliation> getAffiliations(String version)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        MUCLightGetAffiliationsIQ mucLightGetAffiliationsIQ = new MUCLightGetAffiliationsIQ(room, version);
+        StanzaFilter responseFilter = new IQReplyFilter(mucLightGetAffiliationsIQ, connection);
+
+        IQ responseIq = connection.createPacketCollectorAndSend(responseFilter, mucLightGetAffiliationsIQ)
+                .nextResultOrThrow();
+        MUCLightAffiliationsIQ mucLightAffiliationsIQ = (MUCLightAffiliationsIQ) responseIq;
+
+        return mucLightAffiliationsIQ.getAffiliations();
+    }
+
+    /**
+     * Get the MUC Light affiliations.
+     * 
+     * @return the room affiliations
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public HashMap<Jid, MUCLightAffiliation> getAffiliations()
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        return getAffiliations(null);
+    }
+
+    /**
+     * Change the MUC Light affiliations.
+     * 
+     * @param affiliations
+     * @return true if it was successfully done.
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public boolean changeAffiliations(HashMap<Jid, MUCLightAffiliation> affiliations)
```

Comment:
Having this method return `boolean` is not necessary and a bad API design. The return without Exception if successful and throw if not. Note that this also applies to all following methods.